runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -365,7 +366,9 @@ end
 ---@return table Table with map of line number to list of diagnostics.
 ---              Structured: { [1] = {...}, [5] = {.... } }
 ---@private
+---@deprecated
```

Comment:
I think that we should make the deprecation its own commit.

Reply:
The function is already private, so the deprecation doesn’t have any user facing effect.

Reply:
@gpanders it does. `vim.deprecate` prints to `messages` (which isn't the case today).

Reply:
In that case I’d prefer to remove the `vim.deprecate` message completely. It’s a private function so plugins/users use it at their own peril.

Reply:
The function is already listed in `deprecated.txt` so I saw the `@deprecation` and `vim.deprecate` just as following up properly on it.

Reply:
But again, `vim.deprecate` creates a disruptive message that's different from the current experience of only receiving a warning in the code.